vowpalwabbit/pmf_to_pdf.cc

```diff
@@ -103,16 +103,15 @@ pdf_data::~pdf_data()
       ec.l.cb.costs.push_back({cost, j + 1, prob * bandwidth_range, 0.0f});
     }
 
-    auto temp_pd = ec.pred.prob_dist;
-    ec.pred.a_s = data.temp_probs;
+    auto saved_pred = ec.pred;
+    ec.pred.a_s = data.temp_pred_a_s;
 
     base.learn(ec);
 
-    data.temp_probs = ec.pred.a_s;
-    ec.pred.prob_dist = temp_pd;
-    transform(data, ec);
-    data.temp_cb = ec.l.cb;
-    ec.l.cb_cont = temp;
+    data.temp_pred_a_s = ec.pred.a_s;
+    data.temp_lbl_cb = ec.l.cb;
```

Comment:
I see that you have removed the line "transform(data, ec)" here. I remember John mentioned we want the ec.pred to be the correct type even for the learn function. So we need to do transform here as well.